src/gui/folderman.cpp

```diff
@@ -825,32 +830,54 @@ void FolderMan::slotStartScheduledFolderSync()
 
 void FolderMan::slotEtagPollTimerTimeout()
 {
-    ConfigFile cfg;
-    auto polltime = cfg.remotePollInterval();
+    const auto folderMapValues = _folderMap.values();
+    QList<Folder *> foldersToRun;
 
-    for (Folder *f : qAsConst(_folderMap)) {
-        if (!f) {
-            continue;
-        }
-        if (f->isSyncRunning()) {
-            continue;
-        }
-        if (_scheduledFolders.contains(f)) {
-            continue;
-        }
-        if (_disabledFolders.contains(f)) {
-            continue;
-        }
-        if (f->etagJob() || f->isBusy() || !f->canSync()) {
-            continue;
-        }
-        if (f->msecSinceLastSync() < polltime) {
-            continue;
-        }
-        QMetaObject::invokeMethod(f, "slotRunEtagJob", Qt::QueuedConnection);
+    // Some folders need not to be checked because they use the push notifications
+    std::copy_if(folderMapValues.begin(), folderMapValues.end(), foldersToRun.begin(), [&](Folder *folder) {
+        const auto account = folder->accountState()->account();
+        const auto &capabilities = account->capabilities();
```

Comment:
But does auto without & not create a copy of the object?

Reply:
Because capabilities() return a reference, not a pointer. Or does the const ensure it will not make any copy?

Reply:
*sigh* Well, if you ask me, capabilities() shouldn't return a const-ref in the first place. And yes it'll copy but said copy is very cheap.

Reply:
Out of curiosity: Why shouldn't return it a const ref?

Reply:
In term of API design this is almost always a bad idea, it locks you to the actual implementation of the function. It couldn't return a local variable anymore it necessarily have to return a member variable or something else which lifetimes exceeds the lifetime of the caller. Very often this is done for dubious micro-optimization reasons.

Since C++ has no explicit lifetime handling this is seriously dangerous business.

To me it's: by default return by value, evaluate return by reference *only* if it has been *measured* as a performance bottleneck (it very rarely is in my experience apart from very specific high performance computation situations).